# [<](2020-03-24.md) 2020-03-25 [>](2020-03-26.md)

177,750 events, 93,618 push events, 148,960 commit messages, 12,657,304 characters


## [rays1987/covid19-review](https://github.com/rays1987/covid19-review)@[70c1cc3c79...](https://github.com/rays1987/covid19-review/commit/70c1cc3c79cf0760bad731022e782d62b34e6f78)
#### Wednesday 2020-03-25 22:39:31 by Sandipan Ray

Systems biology approaches for understanding SARS-CoV-2 pathogenesis

Systems biology provides a cross-disciplinary analytical platform integrating the different omics (genomics, transcriptomics, proteomics, metabolomics, and other omics approaches), bioinformatics, and computational strategies. These cutting-edge research approaches have enormous potential to study the complexity of biological systems and human diseases (Hood and Tian, 2012, PMID: 23084773). Over the last decade, systems biology approaches are used vastly to study the pathogenesis of diverse types of life-threatening acute and chronic infectious diseases (Eckhardt et al., 2020, PMID: 32060427). Omics-based studies also provided meaningful information regarding host immune responses and surrogate protein markers in several viral, bacterial and protozoan infections (Ray et al., 2014, PMID: 24293340).
The complex pathogenesis and clinical manifestations of SARS-CoV-2 infection are not understood adequately yet. A prominent breakthrough in 2019-nCoV research is accomplished with the successful full-length genome sequencing of the pathogen (Wu et al., Nature 2020, PMID: 32015508; Lu et al., Lancet. 2020, PMID: 32007145, and Zhou et al., 2020, PMID: 32015507). Multiple research groups drafted the genome sequence of SARS-CoV-2 from different clinical samples, including bronchoalveolar lavage fluid, blood, oral and anal swabs, and cultured isolates. Importantly, SARS-CoV-2 has significant sequence homology with SARS-CoV (about 79%) and also to some extent with MERS-CoV (about 50%) (Lu et al., Lancet. 2020, PMID: 32007145). However, a higher level of similarity (about 90%) has been observed between SARS-CoV-2 and bat-derived SARS-like coronaviruses (bat-SL-CoVZC45 and bat-SL-CoVZXC21) indicating its possible bat origin (Lu et al., Lancet. 2020, PMID: 32007145; Zhou et al., 2020, PMID: 32015507). 
The genome sequence of the pathogen subsequently allowed its phylogenetic characterization and prediction of its protein expression profile, which is crucial for understanding the pathogenesis and virulence of this novel viral infection.  In a recent computational bioinformatics study, Grifoni et al. predicted potential B and T cell epitopes for SARS-CoV-2, proving some clue regarding which part of the pathogen is recognized by human immune responses (Grifoni et al., 2020, PMID: 32183941). Importantly, overactivation of T cells, caused by an increase of Th17 and high cytotoxicity of CD8 T cells, could be one of the major reasons behind the severe immune injury and fatality associated with SARS-CoV-2 infection (Xu et al., 2020, PMID: 32085846). 
Availability of the genome sequence of SARS-CoV-2 enhanced the possibilities for the subsequent proteome level studies that can provide further mechanistic insights into its complex pathogenesis. Of note, the cryo-electron microscopy structure of the SARS-CoV-2 spike (S) glycoprotein, which plays an important role in the early steps of viral infection, is reported very recently (Wrapp et al., 2020, PMID: 32075877). Even though any comprehensive proteomic analysis of the pathogen or the patients suffering from its infection has not been reported yet, one forthcoming study has demonstrated SARS-CoV-2 infected host cell proteomics using human Caco–2 cells as an infection model (Bojkova et al., 2020, doi:10.21203/rs.3.rs-17218/v1).  The authors observed SARS-CoV-2 induced alterations in multiple vital physiological pathways, including translation, splicing, carbon metabolism and nucleic acid metabolism in the host cells.
There is a high level of sequence homology between SARS-CoV-2 with SARS-CoV, and sera from convalescent SARS-CoV patients can effectively cross-neutralize SARS-CoV-2-S-driven entry (Hoffmann et al., 2020, PMID: 32142651). Consequently, earlier proteome level studies on SARS-CoV can also provide some essential information regarding the new pathogen (Chen et al., 2004, PMID: 15572443; He et al., 2004, PMID: 15020242). Considering the paucity of omics-level big data sets on for SARS-CoV-2 till now, the existing data hubs that contain adequate information for the other coronaviruses such as UniProt, NCBI Genome Database, The Immune Epitope Database and Analysis Resource (IEDB), The Virus Pathogen Resource (ViPR) could be the useful resources for computational and bioinformatics research on SARS-CoV-2.

---
## [fish2000/CLU](https://github.com/fish2000/CLU)@[ea6533a5a6...](https://github.com/fish2000/CLU/commit/ea6533a5a6fd396a8051116f8f5c2d6dfc40e366)
#### Wednesday 2020-03-25 22:39:36 by Alexander Böhn

Hoisted all of the “types” namespace init-code out of module level
... to wit: that involved taking a bunch of imperative directives
    out of the “clu.typespace” modules’ ‘__init__.py’ file and then
    sticking them back in there wrapped in a function that gets
    called once, at module level, to assign its return value to the
    “types” namespace.
... this lets the imports, some of which are quite fragile with
    circularity concerns, get thrown into the function’s execution
    block, and thus isolated
... nevertheless there were a lot of tweaks that had to be made
    following this change, as a lot of my code did the sloppy and
    expedient thing of importing a thing or two, here and there,
    from “clu.typespace” instead of wherever the fuck the thing
    came from in the first place… that is one thing about CLU and
    all of its module exporters – how you can’t have a some thing
    “Thing” and export it from two places, e.g. “clu.things.Thing”
    *or* “clu.dst.thingamabobs.abc.Thing” is okay but not *both*.
... see yeah if you *were* to export “Thing” from both places,
    calls like `moduleof(Thing)` and its ilk would end up being
    nondeterministic.
... okay so like a photon passing a black hole, this commit note
    has veered off on quite a serious fucking tangent. But I do
    see now that my module look-up stuff (as seen when one executes
    `python -m clu`) should find any such duplicates – even when
    implicit, like if “clu.things.Thing” is only exported the once
    but is imported from elsewhere in some other spot – and flag
    the fuck out of them in the reddest of ANSI red text
... so are we cowabunga on this? yes, we’re cowabunga.

---
## [microsoft/terminal](https://github.com/microsoft/terminal)@[680577f55c...](https://github.com/microsoft/terminal/commit/680577f55c6927ae4f1a63ae396aff6b9b0fae66)
#### Wednesday 2020-03-25 22:47:47 by Michael Niksa

Update til::bitmap to use dynamic_bitset<> + libpopcnt (#5092)

This commit replaces `std::vector<bool>` with `dynamic_bitset<>` by
@pinam45 (https://github.com/pinam45/dynamic_bitset) and with
`libpopcnt` for high-performance bit counting by @kimwalisch
(https://github.com/kimwalisch/libpopcnt).

* [x] In support of performance, incremental rendering, and Terminal
  "not speed enough" as well as my sanity relative to
  `std::vector<bool>`
* [x] Tests updated and passed.
* [x] `LICENSE`, `NOTICE`, and provenance files updated.
* [x] I'm a core contributor. I discussed it with @DHowett-MSFT and
  cleared the licensing checks before pulling this in.

## Details `std::vector<bool>` provided by the Microsoft VC Runtime is
incapable of a great many things. Many of the methods you come to expect
off of `std::vector<T>` that are dutifully presented through the `bool`
variant will spontaneously fail at some future date because it decides
you allocated, resized, or manipulated the `vector<bool>` specialization
in an unsupported manner. Half of the methods will straight up not work
for filling/resizing in bulk. And you will tear your hair out as it will
somehow magically forget the assignment of half the bits you gave it
part way through an iteration then assert out and die.

As such, to preserve my sanity, I searched for an alternative. I came
across the self-contained header-only library `dynamic_bitset` by
@pinam45 which appears to do as much of `boost::dynamic_bitset` as I
wanted, but without including 400kg of boost libraries. It also has a
nifty optional dependency on `libpopcnt` by @kimwalisch that will use
processor-specific extensions for rapidly counting bits. @DHowett-MSFT
and I briefly discussed how nice `popcnt` would have been on
`std::vector<bool>` last week... and now we can have it. (To be fair, I
don't believe I'm using it yet... but we'll be able to easily dial in
`til::bitmap` soon and not worry about a performance hit if we do have
to walk bits and count them thanks to `libpopcnt`.)

This PR specifically focuses on swapping the dependencies out and
ingesting the new libraries. We'll further tune `til::bitmap` in future
pulls as necessary.

## Validation
* [x] Ran the automated tests for bitmap.
* [x] Ran the terminal manually and it looks fine still.

---
## [mayralares/Portfolio2](https://github.com/mayralares/Portfolio2)@[39d85d3e58...](https://github.com/mayralares/Portfolio2/commit/39d85d3e58892038f0107d216540ea34b974ce35)
#### Wednesday 2020-03-25 23:34:19 by mayralares

(Update3) Piano Tiles of Love and Despair

Changed the 'Make it Right' themed p5.js screenshot to the actual project with which you can interact with.

---

# [<](2020-03-24.md) 2020-03-25 [>](2020-03-26.md)

